nginx/public/node/frontend/public/js/app/manuscript-detail/folio/ChantRecordView.js

```diff
@@ -53,6 +115,49 @@ function CustomSplit(str, delimiter, removeEmptyItems) {
 	return result;
 }
 
+function audioStopReset(MIDI){
+	var audioCxt = MIDI.getContext();
+	audioCxt.close();
+	for (var i = 0; i < MIDI.sources.length; i++){
+		MIDI.sources[i].disconnect();
+	}
+	var newAudioCxt = new window.AudioContext();
+	MIDI.setContext(newAudioCxt);
+	$('.btnPlay').html("Play Audio");
+	$('.btnPlay').attr("disabled", false);
+}
+
+// A basic function to handle flats.
+// Flats are applied to B's that immediately follow a flat
+// along with any B's in the same text syllable, unless
+// a natural cancels it.
+function handleFlats (inputStr){
+	var flatReplDict = {};
+	flatReplDict["y"] = ["b","t"];
+	flatReplDict["i"] = ["j","u"];
+	flatReplDict["z"] = ["q","v"];
```

Comment:
can you add comments to each of these lines, something like `B4 -> Bb4`?

Reply:
yes